examples/typescript/aws-move/main.ts

```diff
@@ -176,9 +176,60 @@ export class ComplexIteratorMoveStack extends TerraformStack {
 }
 // MOVE INTO RESOURCE USING COMPLEX ITERATOR
 
+export class MoveToIDStack extends TerraformStack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    new AwsProvider(this, "aws", {
+      region: "us-west-2",
+    });
+    if (process.env.STEP_2) {
+      new S3Bucket(this, "test-bucket-move-by", {
+        bucket: "test-move-bucket-move-to-id",
+      }).moveToId("aws_s3_bucket.test-bucket-1");
+
+      new S3Bucket(this, "test-bucket-1", {
+        bucket: "test-move-bucket-move-to-id",
+      });
+    }
+    if (process.env.STEP_1) {
+      new S3Bucket(this, "test-bucket-move-by", {
+        bucket: "test-move-bucket-move-to-id",
+      });
+    }
+  }
+}
+
+export class MoveFromIDStack extends TerraformStack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    new AwsProvider(this, "aws", {
+      region: "us-west-2",
+    });
+    if (process.env.STEP_2) {
+      new S3Bucket(this, "test-bucket-move-by", {
+        bucket: "test-move-bucket-move-from-id",
+      }).hasMoved();
```

Comment:
curious: Would an alternative be to remove this code? Or does it has to exist until the move is done?

Reply:
Would be a completely valid way to do it. My main reasoning at the time for including it was to keep some semblance of consistency between all the ways of moving resources. Or in other words, it felt odd to have users delete resource definitions in this workflow where they wouldn't in others.

Must admit that I was thinking too narrowly, as the old resource will most likely be deleted anyways. So the use of `hasMoved` is redundant. 

Going to just remove the convention 👍